contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java

```diff
@@ -0,0 +1,398 @@
+            case STRING:
+              String fieldValue = "";
+              fieldValue = cellValue.formatAsString();
+              if (fieldValue.length() > 1) {
+                fieldValue = fieldValue.substring(1);
+                fieldValue = fieldValue.substring(0, fieldValue.length() - 1);
+              }
+              writeStringColumn(rowWriter, fieldName, fieldValue);
+              break;
+          }
+        }
+      }
+      rowWriter.save();
+      recordCount++;
+    }
+    return true;
+  }
+
+  private void writeStringColumn(TupleWriter rowWriter, String name, String value) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.VARCHAR, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setString(value);
+  }
+
+  private void writeFloat8Column(TupleWriter rowWriter, String name, double value) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.FLOAT8, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setDouble(value);
+  }
+
+  private void writeTimestampColumn(TupleWriter rowWriter, String name, Instant ts) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.TIMESTAMP, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setTimestamp(ts);
+  }
+
+  /**
+   * Validates configuration options
+   */
+  private void verifyConfigOptions() {
+    // Validate the config variables
+    if ((readerConfig.lastColumn < readerConfig.firstColumn) && readerConfig.lastColumn != 0) {
+      throw UserException.validationError().message("Invalid column configuration.  The first column index is greater than the last column index.").build(logger);
+    }
+
+    if (readerConfig.firstColumn < 0) {
+      throw UserException.validationError().message("Invalid value for first column.  Index must be greater than zero.").build(logger);
+    }
+
+    if (readerConfig.lastColumn < 0) {
+      throw UserException.validationError().message("Invalid value for last column.  Index must be greater than zero.").build(logger);
+    }
+
+    if (readerConfig.headerRow > readerConfig.lastRow) {
+      throw UserException.validationError().message("Invalid value for headerRow.  Header row must be less than last row.").build(logger);
+    }
+  }
+
+  @Override
+  public void close() {
+    if (workbook != null) {
+      try {
+        workbook.close();
+      } catch (IOException e) {
+        logger.warn(e.getMessage());
```

Comment:
Please add context to the erorr message: `logger.warn("Error when closing XSSFWorkbook resource: {}, e.getMessage())"`